### ISSUE TITLE
Unknown symbol type IdeaCeylonProject kind: 63

### ISSUE THREAD

**Issue by bjansen** [ceylon/ceylon-ide-intellij#443]
When building ceylon-ide-intellij from within IDEA, I get the following compiler error:

```
Error:ceylon: An exception has occurred in the compiler (ceylon 1.2.3 7f789c7 (Total Internal Reflection)).
Error:ceylon: java.lang.AssertionError: Unknown symbol type org.intellij.plugins.ceylon.ide.ceylonCode.model.IdeaCeylonProject kind: 63
Error:ceylon: 	at com.redhat.ceylon.langtools.tools.javac.comp.Check.checkFlags(Check.java:1134)
Error:ceylon: 	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.visitClassDef(Enter.java:402)
Error:ceylon: 	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:692)
Error:ceylon: 	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.classEnter(Enter.java:263)
Error:ceylon: 	at com.redhat.ceylon.compiler.java.loader.CeylonEnter.classEnter(CeylonEnter.java:325)
Error:ceylon: 	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.classEnter(Enter.java:277)
Error:ceylon: 	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.visitTopLevel(Enter.java:336)
Error:ceylon: 	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:515)
Error:ceylon: 	at com.redhat.ceylon.compiler.java.loader.CeylonEnter.classEnter(CeylonEnter.java:320)
Error:ceylon: 	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.complete(Enter.java:542)
Error:ceylon: 	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.main(Enter.java:527)
Error:ceylon: 	at com.redhat.ceylon.compiler.java.loader.CeylonEnter.main(CeylonEnter.java:233)
Error:ceylon: 	at com.redhat.ceylon.langtools.tools.javac.main.JavaCompiler.enterTrees(JavaCompiler.java:985)
Error:ceylon: 	at com.redhat.ceylon.langtools.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:859)
Error:ceylon: 	at com.redhat.ceylon.compiler.java.tools.LanguageCompiler.compile(LanguageCompiler.java:265)
Error:ceylon: 	at com.redhat.ceylon.compiler.java.launcher.Main.compile(Main.java:651)
Error:ceylon: 	at com.redhat.ceylon.langtools.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:132)
Error:ceylon: 	at com.redhat.ceylon.langtools.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:141)
Error:ceylon: 	at com.redhat.ceylon.compiler.java.tools.CeyloncTaskImpl.call2(CeyloncTaskImpl.java:63)
Error:ceylon: 	at com.redhat.ceylon.compiler.java.runtime.tools.impl.JavaCompilerImpl.compile(JavaCompilerImpl.java:106)
Error:ceylon: 	at org.intellij.plugins.ceylon.jps.CeylonBuilder.compile(CeylonBuilder.java:160)
Error:ceylon: 	at org.intellij.plugins.ceylon.jps.CeylonBuilder.build(CeylonBuilder.java:111)
Error:ceylon: 	at org.jetbrains.jps.incremental.IncProjectBuilder.runModuleLevelBuilders(IncProjectBuilder.java:1237)
Error:ceylon: 	at org.jetbrains.jps.incremental.IncProjectBuilder.runBuildersForChunk(IncProjectBuilder.java:911)
Error:ceylon: 	at org.jetbrains.jps.incremental.IncProjectBuilder.buildTargetsChunk(IncProjectBuilder.java:983)
Error:ceylon: 	at org.jetbrains.jps.incremental.IncProjectBuilder.buildChunkIfAffected(IncProjectBuilder.java:870)
Error:ceylon: 	at org.jetbrains.jps.incremental.IncProjectBuilder.buildChunks(IncProjectBuilder.java:696)
Error:ceylon: 	at org.jetbrains.jps.incremental.IncProjectBuilder.runBuild(IncProjectBuilder.java:387)
Error:ceylon: 	at org.jetbrains.jps.incremental.IncProjectBuilder.build(IncProjectBuilder.java:194)
Error:ceylon: 	at org.jetbrains.jps.cmdline.BuildRunner.runBuild(BuildRunner.java:138)
Error:ceylon: 	at org.jetbrains.jps.cmdline.BuildSession.runBuild(BuildSession.java:294)
Error:ceylon: 	at org.jetbrains.jps.cmdline.BuildSession.run(BuildSession.java:125)
Error:ceylon: 	at org.jetbrains.jps.cmdline.BuildMain$MyMessageHandler$1.run(BuildMain.java:232)
Error:ceylon: 	at org.jetbrains.jps.service.impl.SharedThreadPoolImpl$1.run(SharedThreadPoolImpl.java:44)
Error:ceylon: 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
Error:ceylon: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
Error:ceylon: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
Error:ceylon: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
Error:ceylon: 	at java.lang.Thread.run(Thread.java:745)
Error:ceylon: An unknown error occurred in the Java backend. See previous messages for more information.
```

The CLI works fine though. @FroMage do you have any idea what could cause this?

**Comment by FroMage** [comment]
`63` is `Kinds.ERR`, so it's an erroneous symbol. This sort of thing requires a debugger and a look at the `JCTree` in question. It appears to be a class.

**Comment by bjansen** [comment]
Gone, for some reason ¯\\\_(ツ)_/¯

**Comment by bjansen** [comment]
It seems to happen every time `IdeaCeylonProject` is modified (different from the version installed in the IDE?)

**Comment by bjansen** [comment]
Oh, actually it happens in the CLI too, if I provide the same list of files to compile as the one computed by the IDE. 

```
18:27 bastien@mbp2:plugin-ceylon-code (master *)$ ceylonm compile /Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/highlighting/CeylonColorSettingsPage.ceylon /Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/model/IdeaCeylonProject.ceylon /Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/module.ceylon /Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/highlighting/CeylonSyntaxAnnotator.ceylon /Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/model/IdeaModelLoader.ceylon /Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/completion/CustomLookupCellRenderer.ceylon /Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/completion/IdeaCompletionProposal.ceylon /Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/highlighting/CeylonHighlighter.ceylon /Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/annotator/CeylonTypeCheckerAnnotator.ceylon /Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/integrations/studio/AndroidStudioSupport.ceylon /Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/model/CeylonModelManager.ceylon /Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/i
/Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/module.ceylon:5: warning: You import JDK7, which is provided by the JDK8 you are running on, but we cannot check that you are not using any JDK8-specific classes or methods. Upgrade your import to JDK8 if you depend on JDK8 classes or methods.
    shared import java.base "7";
    ^
/Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/module.ceylon:12: warning: You import JDK7, which is provided by the JDK8 you are running on, but we cannot check that you are not using any JDK8-specific classes or methods. Upgrade your import to JDK8 if you depend on JDK8 classes or methods.
    shared import java.desktop "7";
    ^
/Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/module.ceylon:13: warning: You import JDK7, which is provided by the JDK8 you are running on, but we cannot check that you are not using any JDK8-specific classes or methods. Upgrade your import to JDK8 if you depend on JDK8 classes or methods.
    import java.compiler "7";
    ^
/Users/bastien/Dev/ceylon/ceylon-ide-intellij/plugin-ceylon-code/source/org/intellij/plugins/ceylon/ide/ceylonCode/model/IdeaModelLoader.ceylon:9: warning: import is never used: class 'ModalityState' has no local references
    ModalityState
    ^
An exception has occurred in the compiler (ceylon 1.3.1 4a52265 (Now We Try It My Way)).
java.lang.AssertionError: Unknown symbol type org.intellij.plugins.ceylon.ide.ceylonCode.model.CeylonModelManager kind: 63
	at com.redhat.ceylon.langtools.tools.javac.comp.Check.checkFlags(Check.java:1134)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.visitClassDef(Enter.java:402)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:692)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.classEnter(Enter.java:263)
	at com.redhat.ceylon.compiler.java.loader.CeylonEnter.classEnter(CeylonEnter.java:325)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.classEnter(Enter.java:277)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.visitTopLevel(Enter.java:336)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:515)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.classEnter(Enter.java:263)
	at com.redhat.ceylon.compiler.java.loader.CeylonEnter.classEnter(CeylonEnter.java:320)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.classEnter(Enter.java:277)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.complete(Enter.java:542)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.main(Enter.java:527)
	at com.redhat.ceylon.compiler.java.loader.CeylonEnter.main(CeylonEnter.java:233)
	at com.redhat.ceylon.langtools.tools.javac.main.JavaCompiler.enterTrees(JavaCompiler.java:985)
	at com.redhat.ceylon.langtools.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:859)
	at com.redhat.ceylon.compiler.java.tools.LanguageCompiler.compile(LanguageCompiler.java:265)
	at com.redhat.ceylon.compiler.java.launcher.Main.compile(Main.java:651)
	at com.redhat.ceylon.compiler.java.launcher.Main.compile(Main.java:563)
	at com.redhat.ceylon.compiler.java.launcher.Main.compile(Main.java:555)
	at com.redhat.ceylon.compiler.java.launcher.Main.compile(Main.java:544)
	at com.redhat.ceylon.compiler.CeylonCompileTool.run(CeylonCompileTool.java:625)
	at com.redhat.ceylon.common.tools.CeylonTool.run(CeylonTool.java:524)
	at com.redhat.ceylon.common.tools.CeylonTool.execute(CeylonTool.java:405)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.redhat.ceylon.launcher.Launcher.runInJava7Checked(Launcher.java:115)
	at com.redhat.ceylon.launcher.Launcher.run(Launcher.java:41)
	at com.redhat.ceylon.launcher.Launcher.run(Launcher.java:34)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.redhat.ceylon.launcher.Bootstrap.runInternal(Bootstrap.java:139)
	at com.redhat.ceylon.launcher.Bootstrap.run(Bootstrap.java:93)
	at com.redhat.ceylon.launcher.Bootstrap.main(Bootstrap.java:85)
ceylon compile: Fatal error: The compiler exited abnormally (4) due to a bug in the compiler.
Please report it:
 https://github.com/ceylon/ceylon/issues/new
Please include:

* the stacktrace printed below
* a description of what you were trying to compile.

Thank you!
com.redhat.ceylon.compiler.CompilerBugException: Bug
	at com.redhat.ceylon.compiler.CeylonCompileTool.handleExitCode(CeylonCompileTool.java:644)
	at com.redhat.ceylon.compiler.CeylonCompileTool.run(CeylonCompileTool.java:626)
	at com.redhat.ceylon.common.tools.CeylonTool.run(CeylonTool.java:524)
	at com.redhat.ceylon.common.tools.CeylonTool.execute(CeylonTool.java:405)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.redhat.ceylon.launcher.Launcher.runInJava7Checked(Launcher.java:115)
	at com.redhat.ceylon.launcher.Launcher.run(Launcher.java:41)
	at com.redhat.ceylon.launcher.Launcher.run(Launcher.java:34)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.redhat.ceylon.launcher.Bootstrap.runInternal(Bootstrap.java:139)
	at com.redhat.ceylon.launcher.Bootstrap.run(Bootstrap.java:93)
	at com.redhat.ceylon.launcher.Bootstrap.main(Bootstrap.java:85)
Caused by: java.lang.AssertionError: Unknown symbol type org.intellij.plugins.ceylon.ide.ceylonCode.model.CeylonModelManager kind: 63
	at com.redhat.ceylon.langtools.tools.javac.comp.Check.checkFlags(Check.java:1134)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.visitClassDef(Enter.java:402)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:692)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.classEnter(Enter.java:263)
	at com.redhat.ceylon.compiler.java.loader.CeylonEnter.classEnter(CeylonEnter.java:325)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.classEnter(Enter.java:277)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.visitTopLevel(Enter.java:336)
	at com.redhat.ceylon.langtools.tools.javac.tree.JCTree$JCCompilationUnit.accept(JCTree.java:515)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.classEnter(Enter.java:263)
	at com.redhat.ceylon.compiler.java.loader.CeylonEnter.classEnter(CeylonEnter.java:320)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.classEnter(Enter.java:277)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.complete(Enter.java:542)
	at com.redhat.ceylon.langtools.tools.javac.comp.Enter.main(Enter.java:527)
	at com.redhat.ceylon.compiler.java.loader.CeylonEnter.main(CeylonEnter.java:233)
	at com.redhat.ceylon.langtools.tools.javac.main.JavaCompiler.enterTrees(JavaCompiler.java:985)
	at com.redhat.ceylon.langtools.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:859)
	at com.redhat.ceylon.compiler.java.tools.LanguageCompiler.compile(LanguageCompiler.java:265)
	at com.redhat.ceylon.compiler.java.launcher.Main.compile(Main.java:651)
	at com.redhat.ceylon.compiler.java.launcher.Main.compile(Main.java:563)
	at com.redhat.ceylon.compiler.java.launcher.Main.compile(Main.java:555)
	at com.redhat.ceylon.compiler.java.launcher.Main.compile(Main.java:544)
	at com.redhat.ceylon.compiler.CeylonCompileTool.run(CeylonCompileTool.java:625)
	... 16 more
```

What's weird is that the compiler complains that `CeylonModelManager` is unknown, but it's one of the file that are passed to the compiler.

**Comment by bjansen** [comment]
@FroMage @tombentley I believe this is a bug in the compiler, wdyt?

* go to `ceylon-ide-intellij/plugin-ceylon-code`
* build the whole module with `ceylon compile`, it works fine
* do an incremental compilation with `ceylon compile source/org/intellij/plugins/ceylon/ide/ceylonCode/model/IdeaCeylonProject.ceylon`, and observe the error above

**Comment by FroMage** [comment]
I confirm that my fix for ceylon/ceylon#6533 does not let me run into the problem you described in the previous comment, so it's fixed.

**Comment by bjansen** [comment]
Looks like it is working fine now, thanks!
